ibm/service/power/resource_ibm_pi_route.go

```diff
@@ -0,0 +1,332 @@
+				return flex.ResourcePowerUserTagsCustomizeDiff(diff)
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_Action: {
+				Default:      Deliver,
+				Description:  "Action for route.",
+				Optional:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{Deliver}, false),
+			},
+			Arg_Advertise: {
+				Default:      Enable,
+				Description:  "Indicates if the route is advertised. Enum: [\"enable\", \"disable\"].",
+				Optional:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{Enable, Disable}, false),
+			},
+			Arg_Destination: {
+				Description: "Destination of route.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_DestinationType: {
+				Default:      IPV4_Address,
+				Description:  "The destination type.",
+				Optional:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{IPV4_Address}, false),
+			},
+			Arg_Enabled: {
+				Default:     true,
+				Description: "Indicates if the route should be enabled in the fabric.",
+				Optional:    true,
+				Type:        schema.TypeBool,
+			},
+			Arg_Name: {
+				Description: "Name of the route.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_NextHop: {
+				Description: "The next hop.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_NextHopType: {
+				Default:      IPV4_Address,
+				Description:  "The next hop type.",
+				Optional:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{IPV4_Address}, false),
+			},
+			Arg_UserTags: {
+				Computed:    true,
+				Description: "The user tags attached to this resource.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Set:         schema.HashString,
+				Type:        schema.TypeSet,
+			},
+
+			// Attributes
+			Attr_CRN: {
+				Computed:    true,
+				Description: "The CRN of this resource.",
+				Type:        schema.TypeString,
+			},
+			Attr_RouteID: {
+				Computed:    true,
+				Description: "The unique route ID.",
+				Type:        schema.TypeString,
+			},
+			Attr_State: {
+				Computed:    true,
+				Description: "The state of the route.",
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceIBMPIRouteCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	action := d.Get(Arg_Action).(string)
+	advertiseExternally := d.Get(Arg_Advertise).(string)
```

Comment:
Please rename these advertise

```suggestion
	advertise := d.Get(Arg_Advertise).(string)
```